### PR TITLE
TM Provider Gateway 

### PR DESCRIPTION
**Pull request by Didainius**
This PR adds resources and data source `vcd_tm_provider_gateway` for managing TM Provider Gateways. In addition, there is a new data source `vcd_tm_tier0_gateway` for lookup of NSX-T Tier 0 Gateways

### CODE REVIEW

File: vcd/datasource_vcd_tm_tier0_gateway.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+			"already_imported": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: fmt.Sprintf("Defines if the T0 is already imported of %s", labelTmTier0Gateway),
+			},
+		},
+	}
+}
+
+func resourceVcdTmTier0GatewayRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Comment (adambarreiro) [inline vcd/datasource_vcd_tm_tier0_gateway.go:49]: Just a nit, but given there's no resource
```suggestion
func datasourceVcdTmTier0GatewayRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

File: website/vcd.erb [hunk 2]
```diff
@@ -517,6 +517,12 @@
             <li<%= sidebar_current("docs-vcd-data-source-tm-ip-space") %>>
               <a href="/docs/providers/vcd/d/tm_ip_space.html">vcd_tm_ip_space</a>
             </li>
+            <li<%= sidebar_current("docs-vcd-data-source-tm-provider-gateway") %>>
+              <a href="/docs/providers/vcd/r/tm_provider_gateway.html">vcd_tm_provider_gateway</a>
```
Comment (adambarreiro) [inline website/vcd.erb:521]:
```suggestion
              <a href="/docs/providers/vcd/d/tm_provider_gateway.html">vcd_tm_provider_gateway</a>
```

File: website/vcd.erb [hunk 3]
```diff
@@ -517,6 +517,12 @@
...
+            </li>
+            <li<%= sidebar_current("docs-vcd-data-source-tm-tier0-gateway") %>>
+              <a href="/docs/providers/vcd/r/tm_tier0_gateway.html">vcd_tm_tier0_gateway</a>
```
Comment (adambarreiro) [inline website/vcd.erb:524]:
```suggestion
              <a href="/docs/providers/vcd/d/tm_tier0_gateway.html">vcd_tm_tier0_gateway</a>
```

File: website/docs/r/tm_provider_gateway.html.markdown [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) A name for Provider Gateway
+* `description` - (Optional) A description for Provider Gateway
+* `region_id` - (Required) A Region ID for
```
Comment (adambarreiro) [inline website/docs/r/tm_provider_gateway.html.markdown:50]: Incomplete statement here

File: website/docs/d/tm_provider_gateway.html.markdown [hunk 5]
```diff
@@ -0,0 +1,39 @@
+---
+layout: "vcd"
+page_title: "VMware Cloud Director: vcd_tm_provider_gateway"
+sidebar_current: "docs-vcd-data-source-tm-tier0-gateway"
```
Comment (adambarreiro) [inline website/docs/d/tm_provider_gateway.html.markdown:4]:
```suggestion
sidebar_current: "docs-vcd-data-source-tm-provider-gateway"
```

**Review by adambarreiro — CHANGES_REQUESTED** [review]
Test passed! Just a few things

File: vcd/datasource_vcd_tm_tier0_gateway.go (hunk 1 above)
Comment (Didainius) [reply vcd/datasource_vcd_tm_tier0_gateway.go:49]: Thanks!

File: website/vcd.erb (hunk 2 above)
Comment (Didainius) [reply website/vcd.erb:521]: Done

File: website/vcd.erb (hunk 3 above)
Comment (Didainius) [reply website/vcd.erb:524]: Done

File: website/docs/r/tm_provider_gateway.html.markdown (hunk 4 above)
Comment (Didainius) [reply website/docs/r/tm_provider_gateway.html.markdown:50]: Fixed.

File: website/docs/d/tm_provider_gateway.html.markdown (hunk 5 above)
Comment (Didainius) [reply website/docs/d/tm_provider_gateway.html.markdown:4]: Fixed.

**Review by adambarreiro — APPROVED** [review]

File: vcd/config_test.go [hunk 6]
```diff
@@ -119,6 +119,7 @@ type TestConfig struct {
 		NsxtManagerUsername string `json:"nsxtManagerUsername"`
 		NsxtManagerPassword string `json:"nsxtManagerPassword"`
 		NsxtManagerUrl      string `json:"nsxtManagerUrl"`
+		NsxtTier0Gateway    string `json:"nsxtTier0Gateway"`
```
Comment (adambarreiro) [inline vcd/config_test.go:122]: I think sample config update with this new field is missing

Comment (Didainius) [reply vcd/config_test.go:122]: Thanks, added!

**Review by lvirbalas — APPROVED** [review]
This is great!

**Pull request merged**